.github/system_tests/test_daemon.py

```diff
@@ -0,0 +1,137 @@
+"""Tests to run with a running daemon."""
+import subprocess
+import sys
+import operator
+import time
+import numpy as np
+
+from aiida import orm
+from aiida.engine.daemon.client import get_daemon_client
+from aiida.orm import load_node
+from aiida.engine import launch
+from aiida.common import exceptions
+
+from aiida_optimize.engines import Bisection
+from aiida_optimize import OptimizationWorkChain
+import sample_processes
+
+TIMEOUTSECS = 4 * 60  # 4 minutes
+
+
+def print_daemon_log():
+    """Print daemon log."""
+    daemon_client = get_daemon_client()
+    daemon_log = daemon_client.daemon_log_file
+
+    print(f"Output of 'cat {daemon_log}':")
+    try:
+        print(subprocess.check_output(
+            ['cat', f'{daemon_log}'],
+            stderr=subprocess.STDOUT,
+        ))
+    except subprocess.CalledProcessError as exception:
+        print(f'Note: the command failed, message: {exception}')
+
+
+def jobs_have_finished(pks):
+    """Check if jobs with given pks have finished."""
+    finished_list = [load_node(pk).is_terminated for pk in pks]
+    node_list = [load_node(pk) for pk in pks]
+    num_finished = len([_ for _ in finished_list if _])
```

Comment:
Nitpick: `sum(finished_list)`?

Reply:
I personally don't quite a fan of sum a bool list. But fine to change if you prefer.